catalyst-gateway/bin/src/db/index/block/certs.rs

```diff
@@ -0,0 +1,259 @@
+            },
+            pallas::ledger::primitives::alonzo::Certificate::PoolRegistration { .. } => {},
+            pallas::ledger::primitives::alonzo::Certificate::PoolRetirement(..) => {},
+            pallas::ledger::primitives::alonzo::Certificate::GenesisKeyDelegation(..) => {},
+            pallas::ledger::primitives::alonzo::Certificate::MoveInstantaneousRewardsCert(_) => {},
+        }
+    }
+
+    /// Index a certificate from a conway transaction.
+    fn index_conway_cert(
+        &mut self, cert: &conway::Certificate, slot_no: u64, txn: i16, block: &MultiEraBlock,
+    ) {
+        #[allow(clippy::match_same_arms)]
+        match cert {
+            pallas::ledger::primitives::conway::Certificate::StakeRegistration(cred) => {
+                // This may not be witnessed, its normal but disappointing.
+                self.stake_address(cred, slot_no, txn, true, false, None, block);
+            },
+            pallas::ledger::primitives::conway::Certificate::StakeDeregistration(cred) => {
+                self.stake_address(cred, slot_no, txn, false, true, None, block);
+            },
+            pallas::ledger::primitives::conway::Certificate::StakeDelegation(cred, pool) => {
+                self.stake_address(cred, slot_no, txn, false, false, Some(pool.to_vec()), block);
+            },
+            pallas::ledger::primitives::conway::Certificate::PoolRegistration { .. } => {},
+            pallas::ledger::primitives::conway::Certificate::PoolRetirement(..) => {},
+            pallas::ledger::primitives::conway::Certificate::Reg(..) => {},
+            pallas::ledger::primitives::conway::Certificate::UnReg(..) => {},
+            pallas::ledger::primitives::conway::Certificate::VoteDeleg(..) => {},
+            pallas::ledger::primitives::conway::Certificate::StakeVoteDeleg(..) => {},
+            pallas::ledger::primitives::conway::Certificate::StakeRegDeleg(..) => {},
+            pallas::ledger::primitives::conway::Certificate::VoteRegDeleg(..) => {},
+            pallas::ledger::primitives::conway::Certificate::StakeVoteRegDeleg(..) => {},
+            pallas::ledger::primitives::conway::Certificate::AuthCommitteeHot(..) => {},
+            pallas::ledger::primitives::conway::Certificate::ResignCommitteeCold(..) => {},
+            pallas::ledger::primitives::conway::Certificate::RegDRepCert(..) => {},
+            pallas::ledger::primitives::conway::Certificate::UnRegDRepCert(..) => {},
+            pallas::ledger::primitives::conway::Certificate::UpdateDRepCert(..) => {},
+        }
+    }
+
+    /// Index the certificates in a transaction.
+    pub(crate) fn index(
+        &mut self, txs: &pallas::ledger::traverse::MultiEraTx<'_>, slot_no: u64, txn: i16,
+        block: &MultiEraBlock,
+    ) {
+        #[allow(clippy::match_same_arms)]
+        txs.certs().iter().for_each(|cert| {
+            match cert {
+                pallas::ledger::traverse::MultiEraCert::NotApplicable => {},
```

Comment:
I think this could be removed with the `#[allow(clippy::match_same_arms)]`

Reply:
These exist, because we will likely need to implement different code for each match arm as we add indexing.
Reducing them would just mean de-reducing them later, which seems unnecessary.  The code isn't difficult to read, not would it impact performance or code generation in any way.